File: UserMenu.java
public class UserMenu {
    static boolean op;
    //Método principal para el menú de Usuario
    public static void menuUser(){
        boolean flag = true;
        do {
            System.out.println("\nSelecciona la opción deseada:\n1) Agregar dinero\n2) Retirar dinero\n3) Consultar Saldo\n4) Salir");
            System.out.print("\nOpción: ");
            int x = Reader.sc.nextInt();
            Reader.sc.nextLine();
            switch (x){
                case 1->{
                    addMoney();
                }
                case 2->{
                    drawMoney();
                }
                case 3 ->{
                    consultAmount();
                }
                case 4->{
                    System.out.println("\nRegresando al menú anterior...");
                    flag = false;
                }
                default -> System.out.println("Error. Selecciona una opción válida ¬¬ -_-");
            }
        }while(flag);
    }
    //Menú para agregar dinero
    public static void addMoney(){
        boolean flag = true;
        do {
            System.out.println("\n1) Agregar\n2) Salir");//Esta función permite al usuario regresar al menú anterior si se arrepiente de su elección.
            System.out.print("\nOpción: ");
            int opc = Reader.sc.nextInt();
            Reader.sc.nextLine();
            switch (opc){
                case 1 ->{
                    System.out.println("\t\nSelecciona tu usuario:\n");
                    int i = 0;
                    for (Employee employee : Bank.getEmployeeList()){
                        System.out.printf("%d) Usuario: %s %s\n", i + 1, employee.getName(), employee.getLastName());
                        i++;
                    }
                    op = true;
                    int y;
                    do {
                        System.out.print("\nRespuesta: ");
                        y = Reader.sc.nextInt();
                        Reader.sc.nextLine();
                        y -= 1;
                        if (y > Bank.getEmployeeList().size()) System.out.println("Error. Ingresa una opción válida ¬¬ -_-");
                        else op = false;
                    } while (op);
                    Employee employee = Bank.getEmployeeList().get(y);
                    System.out.println("\nSelecciona la cuenta:\n");
                    int x = 0;
                    for(BankAccount account:employee.getAccountList()){
                        System.out.printf("%d) Cuenta: %s Tipo: %s\n",x+1,account.getAccountNumber(),account.getType());
                        x++;
                    }
                    op = true;
                    int z;
                    do{
                        System.out.print("\nRespuesta: ");
                        z = Reader.sc.nextInt();
                        Reader.sc.nextLine();
                        z -= 1;
                        if(z > employee.getAccountList().size()) System.out.println("Error. Ingresa una opción válida ¬¬ -_-");
                        else op = false;
                    }while(op);
                    BankAccount account = employee.getAccountList().get(z);
                    System.out.print("\nIngresa el monto: ");
                    double amount = Reader.sc.nextDouble();
                    Reader.sc.nextLine();
                    account.setAmount(amount);
                    System.out.println("\nMonto actualizado/sin cambios. Regresando al menú...");
                }
                case 2 ->{
                    System.out.println("\nRegresando al menú anterior...");
                    flag = false;
                }
                default -> System.out.println("Error. Selecciona una opción válida ¬¬ -_-");
            }
        }while(flag);
    }
    //Menú para retirar dinero
    public static void drawMoney(){
        boolean flag = true;
        do {
            System.out.println("\n1) Retirar\n2) Salir");
            System.out.print("\nOpción: ");
            int opc = Reader.sc.nextInt();
            Reader.sc.nextLine();
            switch (opc){
                case 1 ->{
                    System.out.println("\t\nSelecciona tu usuario:\n");
                    int i = 0;
                    for (Employee employee : Bank.getEmployeeList()){
                        System.out.printf("%d) Usuario: %s %s\n", i + 1, employee.getName(), employee.getLastName());
                        i++;
                    }
                    op = true;
                    int y;
                    do {
                        System.out.print("\nRespuesta: ");
                        y = Reader.sc.nextInt();
                        Reader.sc.nextLine();
                        y -= 1;
                        if (y > Bank.getEmployeeList().size()) System.out.println("Error. Ingresa una opción válida ¬¬ -_-");
                        else op = false;
                    } while (op);
                    Employee employee = Bank.getEmployeeList().get(y);
                    System.out.println("\nSelecciona la cuenta:\n");
                    int x = 0;
                    for(BankAccount account:employee.getAccountList()){
                        System.out.printf("%d) Cuenta: %s Tipo: %s Saldo: $ %.2f\n",x+1,account.getAccountNumber(),account.getType(),account.getAmount());
                        x++;
                    }
                    op = true;
                    int z;
                    do{
                        System.out.print("\nRespuesta: ");
                        z = Reader.sc.nextInt();
                        Reader.sc.nextLine();
                        z -= 1;
                        if(z > employee.getAccountList().size()) System.out.println("Error. Ingresa una opción válida ¬¬ -_-");
                        else op = false;
                    }while(op);
                    BankAccount account = employee.getAccountList().get(z);
                    System.out.print("\nIngresa el monto: ");
                    double amount = Reader.sc.nextDouble();
                    Reader.sc.nextLine();
                    account.drawMoney(amount);
                    System.out.println("\nMonto actualizado/sin cambios. Regresando al menú...");
                }
                case 2 ->{
                    System.out.println("\nRegresando al menú anterior...");
                    flag = false;
                }
                default -> System.out.println("Error. Selecciona una opción válida ¬¬ -_-");
            }
        }while (flag);
    }
    //Menú para consultar saldo
    public static void consultAmount(){
        boolean flag = true;
        do {
            System.out.println("\n1) Consultar\n2) Salir");
            System.out.print("\nOpción: ");
            int opc = Reader.sc.nextInt();
            Reader.sc.nextLine();
            switch (opc){
                case 1 ->{
                    System.out.println("\t\nSelecciona tu usuario:\n");
                    int i = 0;
                    for (Employee employee : Bank.getEmployeeList()){
                        System.out.printf("%d) Usuario: %s %s\n", i + 1, employee.getName(), employee.getLastName());
                        i++;
                    }
                    op = true;
                    int y;
                    do {
                        System.out.print("\nRespuesta: ");
                        y = Reader.sc.nextInt();
                        Reader.sc.nextLine();
                        y -= 1;
                        if (y > Bank.getEmployeeList().size()) System.out.println("Error. Ingresa una opción válida ¬¬ -_-");
                        else op = false;
                    } while (op);
                    Employee employee = Bank.getEmployeeList().get(y);
                    System.out.println("\nSelecciona la cuenta a consultar:\n");
                    int x = 0;
                    for(BankAccount account:employee.getAccountList()){
                        System.out.printf("%d) Cuenta: %s Tipo: %s\n",x+1,account.getAccountNumber(),account.getType());
                        x++;
                    }
                    op = true;
                    int z;
                    do{
                        System.out.print("\nRespuesta: ");
                        z = Reader.sc.nextInt();
                        Reader.sc.nextLine();
                        z -= 1;
                        if(z > employee.getAccountList().size()) System.out.println("Error. Ingresa una opción válida ¬¬ -_-");
                        else op = false;
                    }while(op);
                    BankAccount account = employee.getAccountList().get(z);
                    System.out.printf("\nSaldo: $ %.2f\n", account.getAmount());
                }
                case 2 ->{
                    System.out.println("\nRegresando al menú principal...");
                    flag = false;
                }
                default -> System.out.println("Error. Selecciona una opción válida ¬¬ -_-");
            }
        }while(flag);
    }
}

File: UserMenu.py
from Bank import Bank

class UserMenu:
    op = False

    @staticmethod
    def menu_user():
        flag = True
        while flag:
            print("\nSelecciona la opción deseada:\n1) Agregar dinero\n2) Retirar dinero\n3) Consultar Saldo\n4) Salir")
            x = int(input("\nOpción: "))
            if x == 1:
                UserMenu.add_money()
            elif x == 2:
                UserMenu.draw_money()
            elif x == 3:
                UserMenu.consult_amount()
            elif x == 4:
                print("\nRegresando al menú anterior...")
                flag = False
            else:
                print("Error. Selecciona una opción válida ¬¬ -_-")

    @staticmethod
    def add_money():
        flag = True
        while flag:
            print("\n1) Agregar\n2) Salir")
            opc = int(input("\nOpción: "))
            if opc == 1:
                print("\t\nSelecciona tu usuario:\n")
                for i, employee in enumerate(Bank.get_employee_list()):
                    print(f"{i + 1}) Usuario: {employee.name} {employee.last_name}")
                op = True
                while op:
                    y = int(input("\nRespuesta: ")) - 1
                    if 0 <= y < len(Bank.get_employee_list()):
                        op = False
                    else:
                        print("Error. Ingresa una opción válida ¬¬ -_-")
                employee = Bank.get_employee_list()[y]
                print("\nSelecciona la cuenta:\n")
                for x, account in enumerate(employee.account_list):
                    print(f"{x + 1}) Cuenta: {account.account_number} Tipo: {account.type}")
                op = True
                while op:
                    z = int(input("\nRespuesta: ")) - 1
                    if 0 <= z < len(employee.account_list):
                        op = False
                    else:
                        print("Error. Ingresa una opción válida ¬¬ -_-")
                account = employee.account_list[z]
                amount = float(input("\nIngresa el monto: "))
                account.set_amount(amount)
                print("\nMonto actualizado/sin cambios. Regresando al menú...")
            elif opc == 2:
                print("\nRegresando al menú anterior...")
                flag = False
            else:
                print("Error. Selecciona una opción válida ¬¬ -_-")

    @staticmethod
    def draw_money():
        flag = True
        while flag:
            print("\n1) Retirar\n2) Salir")
            opc = int(input("\nOpción: "))
            if opc == 1:
                print("\t\nSelecciona tu usuario:\n")
                for i, employee in enumerate(Bank.get_employee_list()):
                    print(f"{i + 1}) Usuario: {employee.name} {employee.last_name}")
                op = True
                while op:
                    y = int(input("\nRespuesta: ")) - 1
                    if 0 <= y < len(Bank.get_employee_list()):
                        op = False
                    else:
                        print("Error. Ingresa una opción válida ¬¬ -_-")
                employee = Bank.get_employee_list()[y]
                print("\nSelecciona la cuenta:\n")
                for x, account in enumerate(employee.account_list):
                    print(f"{x + 1}) Cuenta: {account.account_number} Tipo: {account.type} Saldo: $ {account.amount:.2f}")
                op = True
                while op:
                    z = int(input("\nRespuesta: ")) - 1
                    if 0 <= z < len(employee.account_list):
                        op = False
                    else:
                        print("Error. Ingresa una opción válida ¬¬ -_-")
                account = employee.account_list[z]
                amount = float(input("\nIngresa el monto: "))
                account.draw_money(amount)
                print("\nMonto actualizado/sin cambios. Regresando al menú...")
            elif opc == 2:
                print("\nRegresando al menú anterior...")
                flag = False
            else:
                print("Error. Selecciona una opción válida ¬¬ -_-")

    @staticmethod
    def consult_amount():
        flag = True
        while flag:
            print("\n1) Consultar\n2) Salir")
            opc = int(input("\nOpción: "))
            if opc == 1:
                print("\t\nSelecciona tu usuario:\n")
                for i, employee in enumerate(Bank.get_employee_list()):
                    print(f"{i + 1}) Usuario: {employee.name} {employee.last_name}")
                op = True
                while op:
                    y = int(input("\nRespuesta: ")) - 1
                    if 0 <= y < len(Bank.get_employee_list()):
                        op = False
                    else:
                        print("Error. Ingresa una opción válida ¬¬ -_-")
                employee = Bank.get_employee_list()[y]
                print("\nSelecciona la cuenta a consultar:\n")
                for x, account in enumerate(employee.account_list):
                    print(f"{x + 1}) Cuenta: {account.account_number} Tipo: {account.type}")
                op = True
                while op:
                    z = int(input("\nRespuesta: ")) - 1
                    if 0 <= z < len(employee.account_list):
                        op = False
                    else:
                        print("Error. Ingresa una opción válida ¬¬ -_-")
                account = employee.account_list[z]
                print(f"\nSaldo: $ {account.amount:.2f}")
            elif opc == 2:
                print("\nRegresando al menú principal...")
                flag = False
            else:
                print("Error. Selecciona una opción válida ¬¬ -_-")
